### ISSUE TITLE
relative uri not support

### ISSUE THREAD

**Issue by acgleader** [flutterchina/dio#1309]
### New Issue Checklist

- [ ] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |      |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |      |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |      |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps
I have nginx backend, and it recives uri like /login, /regist，but dio not support，i need like following code:

  /// generate uri
  Uri get uri {
    var _url = path;
    if (baseURL.isNotEmpty) { // add line for check baseURL is not empty
      if (!_url.startsWith(RegExp(r'https?:'))) {
        _url = baseUrl + _url;
        var s = _url.split(':/');
        _url = s[0] + ':/' + s[1].replaceAll('//', '/');
      }
    }
    var query = Transformer.urlEncodeMap(queryParameters, listFormat);
    if (query.isNotEmpty) {
      _url += (_url.contains('?') ? '&' : '?') + query;
    }
    // Normalize the url.
    return Uri.parse(_url).normalizePath();
  }

**Comment by acgleader** [comment]
use latest version, it's not issue
